src/App.tsx

```diff
@@ -1,93 +1,104 @@
 
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
-        />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = event.target.value.trim();
 
-        <ul className="todo-list" data-cy="todoList">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view" />
-              <label htmlFor="toggle-view">asdfghj</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+    setValue(newValue === '' ? '' : event.target.value);
+  };
 
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-completed" />
-              <label htmlFor="toggle-completed">qwertyuio</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const handleInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (value.length && event.key === 'Enter') {
+      event.preventDefault();
+      const newTodoTitle = value;
 
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-editing" />
-              <label htmlFor="toggle-editing">zxcvbnm</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+      setTodos([
+        ...todos,
+        {
+          id: +new Date(),
+          title: newTodoTitle,
+          completed: false,
+        },
+      ]);
 
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view2" />
-              <label htmlFor="toggle-view2">1234567890</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+      setValue('');
+    }
+  };
 
-      <footer className="footer">
-        <span className="todo-count" data-cy="todosCounter">
-          3 items left
-        </span>
+  const handleAllCheckboxes = () => {
+    if (isAllCompleted) {
+      setIsAllChecked(false);
+      setTodos(todos.map((todo: Todo) => ({
+        ...todo,
+        completed: false,
+      })));
+    } else {
+      setIsAllChecked(!isAllChecked);
+      setTodos(todos.map((todo: Todo) => ({
+        ...todo,
+        completed: !isAllChecked,
+      })));
+    }
+  };
```

Comment:
you don't need to map items twice. make it once and set completed value either true or false